Jenkinsfile

```diff
@@ -0,0 +1,40 @@
+#!groovy
+
+def nodeLabel = 'docker'
+
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+node(nodeLabel) {
+  stage 'Checkout'
+  checkout scm
+
+  stage 'Build'
+
+  /* Call the maven build. */
+  mvn "clean install -B"
+
+  /* Archive the test results */
+  step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
+
+  /* Archive the build artifacts */
+  step([$class: 'ArtifactArchiver', artifacts: 'target/*.hpi'])
+}
+
+/* Run maven from tool "mvn" */
+void mvn(def args) {
+    /* Get jdk tool. */
+    String jdktool = tool name: "jdk8", type: 'hudson.model.JDK'
+
+    /* Get the maven tool. */
+    def mvnHome = tool name: 'mvn'
+
+    /* Set JAVA_HOME, and special PATH variables. */
+    List javaEnv = ["PATH+JDK=${jdktool}/bin", "JAVA_HOME=${jdktool}"]
+
+    /* Call maven tool with java envVars. */
+    withEnv(javaEnv) {
+      sh "${mvnHome}/bin/mvn ${args}"
```

Comment:
Wouldn't it be safe to include a multi-platform line in the Jenkinsfile already?  Until INFRA-606 is implemented, it will be harmless in the official project infrastructure (no windows machines means no risk that windows machines will be needed).  I already run a combination of Windows and Linux machines in my git plugin development environment already, so I'll see problems first anyway.  

I'd be the only one using that step, but it would be available for me and would let me leave the jobs unrestricted so they could run on any of my nodes.